import-service/src/handlers/importFileParser.ts

```diff
@@ -0,0 +1,73 @@
+import { S3Event } from "aws-lambda";
+import {
+  CopyObjectCommand,
+  DeleteObjectCommand,
+  GetObjectCommand,
+  S3Client,
+} from "@aws-sdk/client-s3";
+
+import { Readable } from "stream";
+import csv from "csv-parser";
+
+import { buildResponse } from "../handlers/libs/utils";
+
+const { IMPORT_SERVICE_AWS_REGION } = process.env;
+
+export const handler = async (event: S3Event) => {
+  const records = event.Records;
+  if (!records.length) throw new Error("Found no record");
+
+  const key = records[0].s3.object.key;
+  const bucketName = event.Records[0].s3.bucket.name;
+
+  const params = {
+    Bucket: bucketName,
+    Key: key,
+  };
+
+  const client = new S3Client({ region: IMPORT_SERVICE_AWS_REGION });
+
+  const getCommand = new GetObjectCommand(params);
+  const deleteCommand = new DeleteObjectCommand(params);
+
+  const parsedKey = key.replace("uploaded", "parsed");
+
+  const copyCommand = new CopyObjectCommand({
+    Bucket: bucketName,
+    Key: parsedKey,
+    CopySource: `${bucketName}/${key}`,
+  });
+
+  try {
+    const file = await client.send(getCommand);
+    const readStream = file.Body;
+
+    if (!(readStream instanceof Readable)) {
+      throw new Error("Failed to read file");
+    }
+
+    await new Promise((resolve) => {
```

Comment:
Hi. You use Promise here - and it is a good idea, but it might be better to extract copying and deleting!! out of 'end' event handler. I tried your approach and for some reasons it didn't work for me. I think it is because of it tries to delete the file it is reading - well, just finished reading, but still working with - and deletion fails. Because of that - I would better wait for the promise which wraps the stream to resolve and then copy and delete.